src/main/java/hudson/plugins/git/GitPublisher.java

```diff
@@ -245,11 +246,9 @@ public boolean perform(AbstractBuild<?, ?> build,
                         //git.push(null);
                     }
                 } catch (FormException e) {
-                    e.printStackTrace(listener.error("Failed to push merge to origin repository"));
-                    return false;
+					throw new AbortException("Failed to push merge to origin repository");
                 } catch (GitException e) {
-                    e.printStackTrace(listener.error("Failed to push merge to origin repository"));
-                    return false;
+					throw new AbortException("Failed to push merge to origin repository");
```

Comment:
This loses any information from the original exception, which may or may not be of value.